common/djangoapps/enrollment/views.py

```diff
@@ -26,7 +29,11 @@
     OAuth2AuthenticationAllowInactiveUser,
     SessionAuthenticationAllowInactiveUser
 )
-from openedx.core.lib.api.permissions import ApiKeyHeaderPermission, ApiKeyHeaderPermissionIsAuthenticated
+from openedx.core.lib.api.permissions import (
+    ApiKeyHeaderPermission,
+    ApiKeyHeaderPermissionIsAuthenticated,
+    IsUserInUrlOrStaff
```

Comment:
Yeah you're right, everything either checks for the user in the URL or does object-based permission. We don't want either of those, so I just went ahead and added a very vanilla "IsStaff" one. Let me know what you think!